From patchwork Tue Sep  6 19:49:03 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Liam Howlett <Liam.Howlett@Oracle.com>
X-Patchwork-Id: 12968108
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id A6B23ECAAA1
	for <linux-mm@archiver.kernel.org>; Tue,  6 Sep 2022 19:50:38 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 153B18D0006; Tue,  6 Sep 2022 15:50:25 -0400 (EDT)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 0AFF28000A; Tue,  6 Sep 2022 15:50:25 -0400 (EDT)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id E1CE78D0010; Tue,  6 Sep 2022 15:50:24 -0400 (EDT)
X-Delivered-To: linux-mm@kvack.org
Received: from relay.hostedemail.com (smtprelay0016.hostedemail.com
 [216.40.44.16])
	by kanga.kvack.org (Postfix) with ESMTP id C57CA8D0006
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 15:50:24 -0400 (EDT)
Received: from smtpin07.hostedemail.com (a10.router.float.18 [10.200.18.1])
	by unirelay08.hostedemail.com (Postfix) with ESMTP id A92E8140333
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 19:50:24 +0000 (UTC)
X-FDA: 79882702368.07.1DD82A0
Received: from mx0b-00069f02.pphosted.com (mx0b-00069f02.pphosted.com
 [205.220.177.32])
	by imf15.hostedemail.com (Postfix) with ESMTP id 481D5A006B
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 19:50:24 +0000 (UTC)
Received: from pps.filterd (m0246631.ppops.net [127.0.0.1])
	by mx0b-00069f02.pphosted.com (8.17.1.5/8.17.1.5) with ESMTP id
 286Id9Ke014060;
	Tue, 6 Sep 2022 19:50:23 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : references : in-reply-to : content-type :
 content-transfer-encoding : mime-version; s=corp-2022-7-12;
 bh=BDRb2xJBiKRjubqVNDS3BBo6JIFYlEZXCh06Gkb2hqI=;
 b=cCgFWYkPdCbbLJD5YSMfSah8ToHSrGy5ceznPxiWxFNikgsO2KCvpjjWiL2OoNTSKB4x
 XaYP9X7DN6ELpDyGiYCWtNl/imqGWT/uleNPuyT7/qSx/EU0z5FnSilA085CiZCaqxSx
 l1MhToFyfppwWr2HpbUW4HyrGWDnsDiYYRa5EcNIEEavwpQgVkrwRp6COYhGT5X4ao2L
 ESkyHVC9I151y5msLazLfUQh2yq49ncuWAZuQloAfRRDeKrcOh60M5kfTAiD42cPAdyU
 W5npGoSLzgv2k2J2hP97IFpLyTy/IoLJZQsxmgUaH1rMdHYiL/wVD3huZ6mr3pOq0VKg XQ==
Received: from phxpaimrmta01.imrmtpd1.prodappphxaev1.oraclevcn.com
 (phxpaimrmta01.appoci.oracle.com [138.1.114.2])
	by mx0b-00069f02.pphosted.com (PPS) with ESMTPS id 3jbwq2eyq6-1
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:50:23 +0000
Received: from pps.filterd
 (phxpaimrmta01.imrmtpd1.prodappphxaev1.oraclevcn.com [127.0.0.1])
	by phxpaimrmta01.imrmtpd1.prodappphxaev1.oraclevcn.com (8.17.1.5/8.17.1.5)
 with ESMTP id 286H2KIk031383;
	Tue, 6 Sep 2022 19:50:22 GMT
Received: from nam11-co1-obe.outbound.protection.outlook.com
 (mail-co1nam11lp2170.outbound.protection.outlook.com [104.47.56.170])
	by phxpaimrmta01.imrmtpd1.prodappphxaev1.oraclevcn.com (PPS) with ESMTPS id
 3jbwc3ajns-2
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:50:22 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=nMj+LdGzk3rkTsvVMx2zXLN2Uv3UmmCSYqFBYJXFx2lK0GEJOwC7D0bVsYm6rxGJR7obvHVbF08AqRkZSZhPR9zk/zSejc/wRCRD/kl13wmIYRKmO6GsglwT2Pn5op24D0Zb1ZICuU7STrCI0y+L/xXoA8Q6FTGzfLGtmN9c8uW6YJ+Q7MfapLPVrNqr/mHfnljRZJyvCsz82KarDjTspPT568EHZTUChpg9jwbQXQqYwsL47SdFGL8G/8GamNBvjTWR9TOp5pCUz2ho2TLKVQwQlOwmcImIr1YWhiWQSfX1YX0rLu7FeOqXRZPlt8+lkbJh1d3gfJcS8cfg8k1KXg==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=BDRb2xJBiKRjubqVNDS3BBo6JIFYlEZXCh06Gkb2hqI=;
 b=Tj/Os55t6BBhmzrAOgCyV8o5iwwAijwdV0ko/gtH1yxMW0BrRY9IFgSumKeSizqMooXbhxuMmwD3qJ+RnyK0tcrYv0HASy3knTSNkWjFZ9N7ODlX7HLbmhlNjcrBhTXEDsvMEJP6ha4iBaZVGtB0FYHPQcyrb/r/9cG99xpbU2V+IYTYXUq93lzQ8X60rAzPfZpnAkoGuSivexXz5PtjnfBF4IoIa6NyXv0hKyMYbQl3KCI7CEMR0oyqmeJAXbZP6KFpKm0YEpykCDrSVvIM1HvURVPa32qUiyoc6Mg8TDwDbrEtFKvyiLcfFaib68Xa0wPd79i++CLjXVUj40Sl8g==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=BDRb2xJBiKRjubqVNDS3BBo6JIFYlEZXCh06Gkb2hqI=;
 b=snSKSE5AqlMJwQ71nj/hNRbykDy/K9n2lHnRP+Wnls2l9XdUD6wq36TNPEF2m8M1vDya0qN3f41yOQkKaJ1QPSFCM5j+f/9fHvtBx/HE3yaFwzOz9XE3/JwgF6RBNZ/u5xP2LQSXog8Kx9pbdWFIScyZh7CutA1fB/RzLzkWMDo=
Received: from SN6PR10MB3022.namprd10.prod.outlook.com (2603:10b6:805:d8::25)
 by CH0PR10MB4876.namprd10.prod.outlook.com (2603:10b6:610:c9::16) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.5588.15; Tue, 6 Sep
 2022 19:50:20 +0000
Received: from SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223]) by SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223%5]) with mapi id 15.20.5588.018; Tue, 6 Sep 2022
 19:50:20 +0000
From: Liam Howlett <liam.howlett@oracle.com>
To: "maple-tree@lists.infradead.org" <maple-tree@lists.infradead.org>,
        "linux-mm@kvack.org" <linux-mm@kvack.org>,
        "linux-kernel@vger.kernel.org"
	<linux-kernel@vger.kernel.org>,
        Andrew Morton <akpm@linux-foundation.org>
CC: Liam Howlett <liam.howlett@oracle.com>, Vlastimil Babka <vbabka@suse.cz>,
        Davidlohr Bueso <dave@stgolabs.net>
Subject: [PATCH v14 61/70] mm/oom_kill: use vma iterators instead of vma
 linked list
Thread-Topic: [PATCH v14 61/70] mm/oom_kill: use vma iterators instead of vma
 linked list
Thread-Index: AQHYwim3MerNwbaMskeb/bUEsoWh5g==
Date: Tue, 6 Sep 2022 19:49:03 +0000
Message-ID: <20220906194824.2110408-62-Liam.Howlett@oracle.com>
References: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
In-Reply-To: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-mailer: git-send-email 2.35.1
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: db3bb110-4a01-4d1f-7930-08da904107e5
x-ms-traffictypediagnostic: CH0PR10MB4876:EE_
x-ms-exchange-senderadcheck: 1
x-ms-exchange-antispam-relay: 0
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 GEwVhTpQ6jpFMRJlMe4zoMxw9jDNhrSg6AIZ5RR1oLbwoGGAm/xcA3atInVGAdL+lLaODs+G9r21aHLzTPWa4sb82X3GkCT7ZhN0BPFgDdgVVhcpHvBFizxJiMUF61k3mWYwP7vrVGrc6v/2dsKOSq5j14ctU0JJuv5CsrQpA8qizScWNj+6QdrMKhwBYRhUB/VUnsTBrbwbr2GnSFy4ScuX0Rc1XEE76K00ofivtVfFGmN7mLSX1YBTh/8qAe3vvmK1ZY3gAjaPTT1f0bzVTkCGJq63ee+tIoEGbCD1J/Gpo1qqDgdo+Ehx+npRkMV8766XSPdkSXIhIPFfbtabEuS/UGLSCgj/oJBha8pJmiHthIzstDqHndyTj9ZCn1c9QFpBPBPfYNztDBnLJWWHxNuMzCocVkxeBsgb0HEMn6gWCbd7jGe1c3swgqTsCmDAgQQGRVdLMNIstbgtspMgHJVr42N1WDcUpUCUFM+M4xQ5YIDx4VYxFmH75cbzVmQGl/86FX7R0TWLuedMc9QyIfDO0dlC10R5kieiQRLtXTEf01IsN2SGExEBkk2nVSRnTuHLF1usNy2i6B/h4SiQ8QX+iREA3dTpu1B2x/JuLb82Uc+SiDQoPzHAlFT6hL8hstDbEncYXcllNZl7xLcp1q+fUpeGS+zMKb+VazfJ1R8seNVyzUNt3Wj4GR2qstFFyLlp5W1qtUE8L8iuHOi0gWNcce8sRk+C54CcwULoVXo25FbtI61qnP8FvaosD0SONQ1GO1cJjr6/m/5102Qs0Q==
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3022.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(13230016)(39860400002)(136003)(366004)(376002)(396003)(346002)(8936002)(6486002)(8676002)(4326008)(91956017)(122000001)(66946007)(66446008)(186003)(66476007)(76116006)(64756008)(66556008)(2616005)(38100700002)(38070700005)(478600001)(83380400001)(36756003)(1076003)(5660300002)(4744005)(41300700001)(71200400001)(6666004)(44832011)(26005)(6506007)(54906003)(316002)(86362001)(2906002)(110136005)(6512007);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata-chunkcount: 1
x-ms-exchange-antispam-messagedata-0: =?iso-8859-1?q?PAqaV4Geeqaw8YHDjVic1mD?=
	=?iso-8859-1?q?hJyVTTaZwokRUKuuSmqsRbJfKxDv4nLclNluvPTiwlWlBV+8mKiRTlOo1Fu3?=
	=?iso-8859-1?q?ZxamKBO6hA2rgcLAcekPO6Po9+Uh5N8boY8neaZ5m2dGc3XkI0fFtWLTi/NU?=
	=?iso-8859-1?q?xobVrGLT+WmajIEpUZv8arQnm4mhxqPpUSZKFmbF9OjEOFSf9XytyT9GtZXQ?=
	=?iso-8859-1?q?D3KhYDNrcYLaCVfrHTxOkbBo8p0CvnsB2lYKJZZAcfjEVzoN6Tc/3p6YAYY1?=
	=?iso-8859-1?q?FPjlciGeRd1kXo0D5KPzausU2ytmaDMY56s35YLPyeKEGCcgXorLeoyW8szJ?=
	=?iso-8859-1?q?x7Nos5ZCz+gUQ7uLEeCYEUMEG6cd2avYBsqW3S7vl7ignEv3Wf9g9k++ZN9c?=
	=?iso-8859-1?q?twGz0obQxDpuJMvigUdDeAH64apIFWmp+mGnE/Da0BaHTrLnpkDHh1WIWhRk?=
	=?iso-8859-1?q?o/VJW50XzJmoz/o/lm6iU29IZnoqjm++5q+rNWu+cInNZHpCQFIFOZYzdQzk?=
	=?iso-8859-1?q?NZ75emN+OaB/LgR8f1ePACW9+Nsfcx87tp0gJOHnkYDkppWBpEbFwn6trMz3?=
	=?iso-8859-1?q?tWprOoxQuJhVA6Eo16MY5AnBjWxCnhja02hNRWUCIvXU1/DlfB7ic/EoqXLY?=
	=?iso-8859-1?q?Vp8KRgghKAkIC03b89hV/fgTCsEw/ZBndEyTYsy+Vtppf0Q8pj+HWMbtb/+t?=
	=?iso-8859-1?q?KNMI1lpWPW+3myy5xUXltTiVaV4/o0WJoqPzqZV7F1ms1u2PzRzLQl1149AQ?=
	=?iso-8859-1?q?SaB8vpkg1hnjnIZ0UF4IMPGFqeSQrKCDjT8uR5aQxdWmqg3CkdTwQnNGzj2i?=
	=?iso-8859-1?q?K+gHfGm/Efp5fcJ6Hh9mdkZDCHJKTSooCVtzWU1PuAvW2ijMPvB2nPdU1CZk?=
	=?iso-8859-1?q?ZZWsX713v44LwA/TujzhZc24L4PSoHcOJCO+HT4jq2WYMaDJ2xdsI7z7x7sK?=
	=?iso-8859-1?q?CPSI9QBO8OPCjLBGvU0mA9IlwoidOdxtaF0ecw2HXX/k1nHj3x5gbtKrp8ru?=
	=?iso-8859-1?q?jh0+su7FwGl8suak9QMzTMaCQKZqRTGXryBar8M5Ty8YhA0qDrJqohbGfP56?=
	=?iso-8859-1?q?ugecl5UFBKNpRmvJVPK1TRQIOIlYDKcefbtXsneV6nnvVkhx9WLsTIOn5DEu?=
	=?iso-8859-1?q?VdejanrnmxNEMduEGnotd9XybJuDMjxczsMW7FWO7hljLW5sxjThTliDmXkC?=
	=?iso-8859-1?q?HjWpkh9364dTSYnRNT1FfnYnQ2zhlaHtiYyV2GcfiunKB9L0RgxKfjLtQbkB?=
	=?iso-8859-1?q?s5peURjpDOlwvvMHszCMmuC9Ihg5ljpQvWtq8bEryRf9V60+gMEZFq6OX1ga?=
	=?iso-8859-1?q?awiT1oaWC69We3986vkcTUnLo9fpjU/nKlKmk+aVbzVruWEWmiW/Tm0WJH1e?=
	=?iso-8859-1?q?zt+q5r8gUApoxt9L+2TKC3n2zX497XXHmb8R5paVZkvvSkN8Wmc7jiQiNYFM?=
	=?iso-8859-1?q?cAY558Ft/RlKGmM9Rwdxdt00+9raH74Epr2Hzed6gXL3i7nSm5V3I2Rp5A1j?=
	=?iso-8859-1?q?TrnUDq2xHDEP4zz/IQ/XsFw79yN0QbPnhfGeZaHuDY++nJZnCQ6GjA/sMN6t?=
	=?iso-8859-1?q?yFxFOy3ia4WQl2vnpROQ5G0NsqKr7uaj2YUscaqaJH7gstxB7SM5vgQcEGrG?=
	=?iso-8859-1?q?khiufYiJnquE1NDTz/CxNbW7CUIi/0cBv5ITxfw=3D=3D?=
MIME-Version: 1.0
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3022.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 db3bb110-4a01-4d1f-7930-08da904107e5
X-MS-Exchange-CrossTenant-originalarrivaltime: 06 Sep 2022 19:49:03.8807
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 Yg7RnwP5+0FKvWWcztbsxwY8puZFTJ64gculb8mQIrPq/KVy7yBLsY3bTU+/buNAVJGv/lJfGOfA5knZMHYMDA==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: CH0PR10MB4876
X-Proofpoint-Virus-Version: vendor=baseguard
 engine=ICAP:2.0.205,Aquarius:18.0.895,Hydra:6.0.528,FMLib:17.11.122.1
 definitions=2022-09-06_09,2022-09-06_02,2022-06-22_01
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 spamscore=0
 malwarescore=0 mlxscore=0
 suspectscore=0 mlxlogscore=999 phishscore=0 adultscore=0 bulkscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2207270000
 definitions=main-2209060091
X-Proofpoint-GUID: dL6t3-Kvgp8-atxZLvndNoYKekXSHKn5
X-Proofpoint-ORIG-GUID: dL6t3-Kvgp8-atxZLvndNoYKekXSHKn5
ARC-Seal: i=2; s=arc-20220608; d=hostedemail.com; t=1662493824; a=rsa-sha256;
	cv=pass;
	b=jU9/hDVEm+fOAnb4EvjTSSxo9QSbJVxT/JoJz58gqFayHz4BE0A6svvOU4+yasgd1HrkgZ
	VGzzSomtT7QX4SYRtQy0skm3Sa7HvMInPaGVFw7uX/cujsys/E9Z9z/ExHoiYQ+2NtdhH2
	bkApbtnxHcpMVv+Gpej2K8Kq9YmCh2o=
ARC-Authentication-Results: i=2;
	imf15.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=cCgFWYkP;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=snSKSE5A;
	spf=pass (imf15.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.177.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com;
	arc=pass ("microsoft.com:s=arcselector9901:i=1")
ARC-Message-Signature: i=2; a=rsa-sha256; c=relaxed/relaxed;
 d=hostedemail.com;
	s=arc-20220608; t=1662493824;
	h=from:from:sender:reply-to:subject:subject:date:date:
	 message-id:message-id:to:to:cc:cc:mime-version:mime-version:
	 content-type:content-type:
	 content-transfer-encoding:content-transfer-encoding:
	 in-reply-to:in-reply-to:references:references:dkim-signature;
	bh=BDRb2xJBiKRjubqVNDS3BBo6JIFYlEZXCh06Gkb2hqI=;
	b=Uc2Ruu4nU3LzUXKuLGnFNRtnnPalp3SvsitE8UBo1BPfm17zDcZPva+zBREMViiQOV5h61
	qO33DslRoUWt4caTFtXwz/3ADuS4tQUmjFOBDDZWHN3H575E1YFbadMHehX30Cr5W9LAbN
	Xc4fX7ZuZyiP7xUPNAnohinhI9JPwHY=
Authentication-Results: imf15.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=cCgFWYkP;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=snSKSE5A;
	spf=pass (imf15.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.177.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com;
	arc=pass ("microsoft.com:s=arcselector9901:i=1")
X-Rspam-User: 
X-Rspamd-Server: rspam06
X-Stat-Signature: p9jpor83mp5g4rgmi9ipy8yu8yewj3ou
X-Rspamd-Queue-Id: 481D5A006B
X-HE-Tag: 1662493824-916623
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

From: "Liam R. Howlett" <Liam.Howlett@Oracle.com>

Use vma iterator in preparation of removing the linked list.

Signed-off-by: Liam R. Howlett <Liam.Howlett@Oracle.com>
Acked-by: Vlastimil Babka <vbabka@suse.cz>
Reviewed-by: Davidlohr Bueso <dave@stgolabs.net>
---
 mm/oom_kill.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mm/oom_kill.c b/mm/oom_kill.c
index c08cffbca0d9..6811aa48a98c 100644
--- a/mm/oom_kill.c
+++ b/mm/oom_kill.c
@@ -513,6 +513,7 @@ bool __oom_reap_task_mm(struct mm_struct *mm)
 {
 	struct vm_area_struct *vma;
 	bool ret = true;
+	VMA_ITERATOR(vmi, mm, 0);
 
 	/*
 	 * Tell all users of get_user/copy_from_user etc... that the content
@@ -522,7 +523,7 @@ bool __oom_reap_task_mm(struct mm_struct *mm)
 	 */
 	set_bit(MMF_UNSTABLE, &mm->flags);
 
-	for (vma = mm->mmap ; vma; vma = vma->vm_next) {
+	for_each_vma(vmi, vma) {
 		if (vma->vm_flags & (VM_HUGETLB|VM_PFNMAP))
 			continue;
 

